server/routes/prisons/categoryGroups/index.ts

```diff
@@ -1,27 +1,26 @@
 import { RequestHandler, Router } from 'express'
 import { ValidationChain } from 'express-validator'
 import { Services } from '../../../services'
-import asyncMiddleware from '../../../middleware/asyncMiddleware'
 import CategoryGroupsController from './categoryGroupsController'
 import SingleCategoryGroupController from './singleCategoryGroupController'
 import AddCategoryGroupController from './addCategoryGroupController'
 
 export default function routes(services: Services): Router {
   const router = Router()
 
-  const get = (path: string | string[], handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
-  const post = (path: string | string[], handler: RequestHandler) => router.post(path, asyncMiddleware(handler))
+  const get = (path: string | string[], handler: RequestHandler) => router.get(path, handler)
+  const post = (path: string | string[], handler: RequestHandler) => router.post(path, handler)
   const postWithValidation = (path: string | string[], validationChain: ValidationChain[], handler: RequestHandler) =>
-    router.post(path, ...validationChain, asyncMiddleware(handler))
+    router.post(path, ...validationChain, handler)
 
   const categoryGroups = new CategoryGroupsController(services.prisonService, services.categoryGroupService)
   const singleCategoryGroup = new SingleCategoryGroupController(services.prisonService, services.categoryGroupService)
   const addCategoryGroup = new AddCategoryGroupController(services.prisonService, services.categoryGroupService)
 
-  get('/prisons/:prisonId([A-Z]{3})/category-groups', categoryGroups.view())
-  get('/prisons/:prisonId([A-Z]{3})/category-groups/add', addCategoryGroup.view())
-  get('/prisons/:prisonId([A-Z]{3})/category-groups/:reference', singleCategoryGroup.view())
-  post('/prisons/:prisonId([A-Z]{3})/category-groups/:reference/delete', singleCategoryGroup.delete())
+  get('/prisons/:prisonId/category-groups', categoryGroups.view())
```

Comment:
Is it worth me creating a middleware fn to check `prisonId` and throwing a 404 if it doesn't match the regexp?

Reply:
It was already a bit inconsistent tbh: `:prisonId` in the path was being validated but not other things (e.g. `:reference`) - I think probably because prison ID is an easy regex; other references have more elaborate formats which were still in flux a bit as we were building this...

Feels like it should be OK without given this app just used internally and all behind Auth? To properly fix this we'd need to make sure all path parameters were validated (with middleware like you say or express-validator)